### PR TITLE
Add example connector

### PR DESCRIPTION
**Pull request by rhaist**
this PR adds an example connector to be used for educational purposes.

It has a corresponding (in progress) guide at:

https://www.notion.so/HowTo-Build-your-first-connector-06b2690697404b5ebc6e3556a1385940

### CODE REVIEW

File: example/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+# Copy the worker
+COPY src /opt/opencti-connector-example
+
+# Install Python modules
+# hadolint ignore=DL3003
+RUN apk --no-cache add git build-base libmagic && \
+    cd /opt/opencti-connector-example && \
+    pip3 install --no-cache-dir git+https://github.com/OpenCTI-Platform/client-python@master && \
+    pip3 install --no-cache-dir -r requirements.txt && \
```
Comment (2xyo) [inline example/Dockerfile:11]: `pycti`  is already defined in `requirements.txt`. Latest stable on pypi in `requirements.txt`  vs latest on git with `pip3 install..`

File: example/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.7-alpine
+
```
Comment (2xyo) [inline example/Dockerfile:2]: Maybe we could add some labels?

```dockerfile
ARG SOURCE_COMMIT
ARG BUILD_DATE

LABEL org.opencontainers.image.title="Example connector" \
    org.opencontainers.image.description="This is the OpenCTI example connector to be used for educational purposes." \
    org.opencontainers.image.authors="Robert Haist <rha@debian.org>" \
    org.opencontainers.image.documentation="https://www.notion.so/HowTo-Build-your-first-connector-06b2690697404b5ebc6e3556a1385940" \
    org.opencontainers.image.licenses="Apache License 2.0" \
    org.opencontainers.image.version="1.0.0" \
    org.opencontainers.image.url="https://github.com/OpenCTI-Platform/connectors/tree/master/example" \
    org.opencontainers.image.revision=$SOURCE_COMMIT \
    org.opencontainers.image.created=$BUILD_DATE
```

With  something like this: 
```console 
$ BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ") 
$ [ -n "$SOURCE_COMMIT" ]       || SOURCE_COMMIT=$(git rev-parse -q HEAD) 
$ docker --build-arg SOURCE_COMMIT=$SOURCE_COMMIT --build-arg BUILD_DATE=$BUILD_DATE ...
```

References:
* https://docs.docker.com/docker-hub/builds/advanced/ 
* https://success.docker.com/article/secure-supply-chain#dockerfilebestpractices
* https://github.com/opencontainers/image-spec/blob/master/annotations.md

File: example/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# Copy the worker
+COPY src /opt/opencti-connector-example
```
Comment (2xyo) [inline example/Dockerfile:4]: Moving the `COPY` at the end avoids invalidating the cache of the intermediate layers if the connector source code change. Example:
```dockerfile
FROM python:3.8-alpine

WORKDIR  /opt/opencti-connector-example

COPY src/requirements.txt  ./

RUN apk --no-cache add build-base && \
    pip3 install --no-cache-dir -r requirements.txt && \
    apk del build-base

COPY src entrypoint.sh ./
 
ENTRYPOINT ["/bin/sh", "-c", "/opt/opencti-connector-example/entrypoint.sh"]

```

File: example/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.7-alpine
```
Comment (2xyo) [inline example/Dockerfile:1]: Maybe you should consider upgrading latest stable 3.8 python ?  

https://docs.python.org/3.8/whatsnew/3.8.html

File: example/src/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+PyYAML
+pika
+pycti
+IPy
```
Comment (2xyo) [inline example/src/requirements.txt:4]: This docker image build is not deterministic/reproducible. Adding the version of dependencies could be a good idea  :smiley:

File: example/Dockerfile (hunk 1 above)
Comment (rhaist) [reply example/Dockerfile:11]: You are totally right. Will be changed.

**Comment by SamuelHassine** [comment]
@rhaist Not ready yet for merging?

**Comment by rhaist** [comment]
No. Still not finished with this. Sorry.

**Comment by rhaist** [comment]
I'll close this for now to avoid confusion and revisit this with v4.x released in the near future.

**Pull request closed without merging**